core/sql/comexe/ComTdbExeUtil.cpp

```diff
@@ -1302,6 +1302,7 @@ void ComTdbExeUtilFastDelete::displayContents(Space * space,
 ComTdbExeUtilHiveTruncate::ComTdbExeUtilHiveTruncate(
      char * tableName,
      ULng32 tableNameLen,
+     char * hiveTableName,
```

Comment:
By "bound at compile time" I mean the Hive table name is chosen at compile time rather than execution time. If the table name were chosen at execution time, we would put it in the TCB (or some other executor data structure) but not the TDB. If we chose the name at execution time, then the query plan could be shared in the query cache without worries.